lib/Wikia/src/Localisation/CachedLCStoreDB.php

```diff
@@ -43,9 +60,33 @@ function get( $code, $key ) {
 			$this->cacheService->set( $cacheKey, $value, static::CACHE_TTL );
 		}
 
+		wfProfileOut( __METHOD__ );
 		return $value;
 	}
 
+	/**
+	 * Return the list of all messages that are kept in localisation cache.
+	 *
+	 * Avoid DB queries for messages that we now they do not exist.
+	 *
+	 * LocalisationCache is in many cases called for not existing messages
+	 * via wfEmptyMsg() and Message::exists() methods
+	 *
+	 * @param string $code
+	 * @return string[]
+	 */
+	private function getAllKeys( $code ) {
+		$cacheKey = $this->getCacheKeyForMessage( $code, __METHOD__ );
+		$keys = $this->cacheService->get( $cacheKey );
+
+		if ( !is_array( $keys ) ) {
```

Comment:
Can $keys ever be an empty array? I suppose it cannot, since something either is cached or not, just asking :)